### PR TITLE
feature: Refine exposed protocol types

### PR DESCRIPTION
**Pull request by aiven-anton**
By refining the exposed protocol types to include unions types for the different kinds of headers, we allow typing at call-sites to work smoother. This also enables using exhaustiveness checking in a place where it wasn't possible before, when matching the type of an entity's header schema. See the replaced NotImplementedError in the diff for reference.

### CODE REVIEW

File: src/kio/static/base.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+__all__ = ("ApiMessage",)
+
+
+class ApiMessage:
```
Comment (aiven-anton) [inline src/kio/static/base.py:4]: `ApiMessage` is moved to a separate module to not cause circular dependency errors, it is also not really an interface/protocol but rather used as a nominal marker for top-level API entities.

We should probably consider redesigning this to be more inline with the rest of the introspection features, and not require an import and subclassing.

File: src/kio/static/protocol.py [hunk 2]
```diff
@@ -1,12 +1,45 @@
+from __future__ import annotations
+
 from typing import ClassVar
 from typing import Protocol
+from typing import TypeAlias
```
Comment (matyaskuti) [inline src/kio/static/protocol.py:5]: Nit, but aren't we missing `isort`? 🤔

Comment (jjaakola-aiven) [reply src/kio/static/protocol.py:5]: Separate PR for this, I presume it would change few files and should be chore of its own to do so.

Comment (matyaskuti) [reply src/kio/static/protocol.py:5]: For sure, adding it doesn't belong here, just stood out :)

Comment (aiven-anton) [reply src/kio/static/protocol.py:5]: Already configured: https://github.com/Aiven-Open/kio/blob/2f9d9cc4c34b42a8481ebf89a79e9b381cb7dbae/pyproject.toml#L137

**Review by jjaakola-aiven — APPROVED** [review]

**Review by jjaakola-aiven — APPROVED** [review]

**Pull request merged**